moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1528,11 +1516,20 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::Attache
           subframe_poses[name] = p;
         }
 
+        // TODO(felixvd): Use relative subframe poses
         // Transform subframes to the link frame
         if (object.object.header.frame_id != object.link_name)
         {
-          const Eigen::Isometry3d& transform = robot_state_->getGlobalLinkTransform(link_model).inverse() *
-                                               getTransforms().getTransform(object.object.header.frame_id);
+          const Eigen::Isometry3d& world_to_object_header_transform =
+              getTransforms().getTransform(object.object.header.frame_id);
+          Eigen::Isometry3d object_header_to_pose_transform;  // TODO(felixvd): Should this assume a neutral
+                                                              // quaternion if all values are 0?
+          PlanningScene::poseMsgToEigen(object.object.pose, object_header_to_pose_transform);
+          const Eigen::Isometry3d& world_to_object_origin_transform =
+              world_to_object_header_transform * object_header_to_pose_transform;
+
+          const Eigen::Isometry3d& transform =
```

Comment:
Consider giving this transform more descriptive name similar to the other transforms.

Reply:
I think this is undefined behavior, transform is a const reference to a temporary (the result of the multiply needs to have a lifetime beyond this statement so it can be used later).  I think what you probably want is just a `const Eigen:isometry3d` and drop the reference.

Reply:
I think `const&` should be fine since it has the ability to keep stuff alive. [https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/).

Reply:
Weird... I wonder why they did that.  I think it would be clearer if we don't depend on `const&` extending the lifetime and just use a normal, on the stack, variable.  The compiler probably generates equally optimal code either way.

Reply:
I changed the reference to a variable where appropriate (including your other comments) and gave it a descriptive name.